src/Aspire.Hosting.Tasks/GetNonExecutableReferences.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+
+namespace Aspire.Hosting.Tasks;
+
+/// <summary>
+/// Gets the references in 'AppProjectTargetFramework' that aren't executable projects.
+/// </summary>
+public sealed class GetNonExecutableReferences : Microsoft.Build.Utilities.Task
+{
+    /// <summary>
+    /// The project target frameworks to check.
+    /// </summary>
+    [Required]
+    public ITaskItem[] AppProjectTargetFramework { get; set; } = [];
+
+    /// <summary>
+    /// The output containing non-executable references.
+    /// </summary>
+    [Output]
+    public ITaskItem[] NonExecutableReferences { get; set; } = [];
+
+    public override bool Execute()
+    {
+        HashSet<ITaskItem> nonExecutableReferences = new HashSet<ITaskItem>();
+
+        foreach (var appProject in AppProjectTargetFramework)
+        {
+            var additionalProperties = appProject.GetMetadata("AdditionalPropertiesFromProject");
+            if (string.IsNullOrEmpty(additionalProperties))
+            {
+                // Skip any projects that don't contain the right metadata
+                continue;
+            }
+
+            var additionalPropertiesXml = XElement.Parse(additionalProperties);
+            foreach (var targetFrameworkElement in additionalPropertiesXml.Elements())
+            {
+                var isExe = targetFrameworkElement.Element("_IsExecutable");
+                if (isExe is null || !string.Equals(isExe.Value, "true", StringComparison.OrdinalIgnoreCase))
```

Comment:
Note this is actually a change in the logic. Before:

```C#
if (isExe != null && !string.Equals(isExe.Value, "true", StringComparison.OrdinalIgnoreCase))
```

This logic is wrong, if the reference doesn't explicitly say `_IsExecutable=true` we want to warn. I'm a bit nervous that we are going to find some project type that isn't setting this property and it is going to regress the experience. But anyone who hits it can just disable the check with `SkipValidateAspireHostProjectResources=true`.